### PR TITLE
Fix message deletion

### PR DESCRIPTION
**Pull request by cketti**
In some situations placeholder messages in K-9 Mail's database haven't been deleted.

### CODE REVIEW

**Comment by basilgello** [comment]
Works for me :) Still need to remove fulltext info about purged messages ;)

> I also noticed that emptying trash adds FTS4 entry (segdir one, specifically):
> 
> ```sql
> INSERT INTO messages_fulltext_segdir VALUES(0,8,0,0,'0 524',X'0005313737373502210000056164646f6e02210001086c77696e6573636802210001026e640221000102726502210000086264656c65746564022100010665636175736502210003046f6d65730221000001630221000114616c6c7265636f7264696e676d656e75686f6f6b02210002067465676f727902210001026f6d02210003066d656e74656402210004026974022100000764656c65746564022100010769726563746c7902210002087363757373696f6e02210001036f6e650221000005656d61696c0221000005676574696402210003076c6162656c69640221000105697468756202210000046861766502210001036f6f6b022100040464617461022100010474747073022100000769686f6f6b696402210001116c6f63616c697a6564737472696e67696402210001017402210000096d656e74696f6e6564022100030575686f6f6b02210000026f6e022100010172022100000470756c6c02210001027672022100000a7234313638323639353002210001086563656976696e6702210003066f7264696e6702210002056c6174656402210002056d6f7665640221000203706c790221000205717565737402210002047475726e0221000204776f726b0221000003746167022100010368697302210001016f02210002056164646f6e022100000476696577022100000477657265022100000478626d630221000003796f75022100');
> ```
> 
> which is left after the message gets purged.

**Comment by cketti** [comment]
> Works for me :) Still need to remove fulltext info about purged messages ;)

I've never looked closely at how SQLite's FTS tables work. But I don't think we're supposed to directly modify that particular table. Can you look into this? I need to work on other things for a while.

**Comment by basilgello** [comment]
> > Works for me :) Still need to remove fulltext info about purged messages ;)
> 
> I've never looked closely at how SQLite's FTS tables work. But I don't think we're supposed to directly modify that particular table. Can you look into this? I need to work on other things for a while.

**tl;dr: Add the folliowing SQL statement at the end of message destroy operations**

`"INSERT INTO messages_fulltext(messages_fulltext) VALUES('optimize')`` 

Detailed : I started looking into segdir issue and foundthis discussion on SQLite3 mailing list:
http://sqlite.1065341.n5.nabble.com/FTS3-4-merge-function-behaviour-when-deleting-rows-td75370.html

It led me to https://www.sqlite.org/fts3.html#mergecmd

Now I am testing the removal of messages using this technique and it looks like deleted messages are really purged form fts4 table.
